src/compiler/output-targets/test/output-targets-dist-custom-elements.spec.ts

```diff
@@ -0,0 +1,151 @@
+import { path } from '@stencil/core/compiler';
+import { mockConfig, mockStencilSystem, mockBuildCtx, mockCompilerCtx, mockModule } from '@stencil/core/testing';
+import type * as d from '../../../declarations';
+import {
+  addCustomElementInputs,
+  generateEntryPoint,
+  getBundleOptions,
+  outputCustomElements,
+} from '../dist-custom-elements';
+import * as outputCustomElementsMod from '../dist-custom-elements';
+import { OutputTargetDistCustomElements } from '../../../declarations';
+import { stubComponentCompilerMeta } from '../../types/tests/ComponentCompilerMeta.stub';
+import { STENCIL_APP_GLOBALS_ID, STENCIL_INTERNAL_CLIENT_ID, USER_INDEX_ENTRY_ID } from '../../bundle/entry-alias-ids';
+
+const setup = () => {
```

Comment:
this test suite isn't precisely comprehensive, I think we'd need the changes in #3127 in order to do full, comprehensive tests for this stuff. But this should move the needle!